### ISSUE TITLE
Invalid JSON generated with rpc.get_config(options={'format': 'json'})

### ISSUE THREAD

**Issue by CMeza99** [Juniper/py-junos-eznc#701]
```python
>>>json.loads(dev.rpc.get_config(options={'format': 'json'}).text)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.5/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 22030 column 17 (char 680605)
```

Looking at the json from `dev.rpc.get_config(options={'format': 'json'}).text` it indeed is missing a `,`:
```vim
22021         "protocols" : [
22022         {
22023             "comment" : "BEGIN: ROUTING-INSTANCES",
22024             "bgp" : [
22025             {
22026                 "apply-groups" : [
22027                 {
22028                     "data" : "BGP_DEFAULTS"
22029                 }
22030                 ]
22031                 "path-selection" : [
22032                 {
22033                     "always-compare-med" : [
22034                     {
22035                         "data" : null
22036                     }
22037                     ]
22038                 }
22039                 ],
```

**Comment by stacywsmith** [comment]
Hi @CMeza99 

Thanks for reporting. This appears to be a Junos bug in the rendering of the configuration in JSON format. If you want to provide more details here I can help in tracking the Junos issue. Alternatively, you can open a JTAC case.

I would need to know:
1) Juniper platform/model
2) Exact version of Junos
3) CLI output of `show configuration | display xml`
4) CLI output of `show configuration | display json`

FYI, you should not have to call `json.loads()` on the text of the RPC reply. Instead, when you request JSON format, PyEZ should perform the `json.loads()` for you and return a native Python data structure.

In this case, that's not happening because the `json.loads()` is raising an exception because the JSON emitted by Junos is invalid. I spoke with @vnitinv about this and he will address this specific PyEZ issue.

**Comment by CMeza99** [comment]
Thank you
1. QFX10008 and QFX10000-RE
2. Junos 15.1X53-D36
3. / 4. I don't think I can post the configuration (per policy and size). But yes, when I run `show configuration | display json` on the switch itself, it returns the invalid json.

I will inquire to how to open a JTAC case from my end and will update this issue once I can.

**Comment by stacywsmith** [comment]
Thanks @CMeza99 
Once you have a case open, you can email me the case number. stacys at juniper dot net

**Comment by stacywsmith** [comment]
While we're working on a resolution to this issue...

You can get the config in XML format and use Etree methods access the data:
https://docs.python.org/2/library/xml.etree.elementtree.html

or if you prefer not to deal with Etree Element objects, you could convert them using jxmlease:
https://github.com/Juniper/jxmlease
jxmlease provides a python data structure which behaves much more like native dicts and lists. The result will look a lot like the data structure produced from `json.loads()` and this solution doesn't depend on the Junos device running a recent version with (working) JSON support.

**Comment by stacywsmith** [comment]
The primary problem in this issue is a Junos bug. The user is addressing the Junos bug through JTAC.

The secondary problem is that the PyEZ behavior for this error condition was confusing and unexpected. This secondary problem has been addressed through #706.
